libraries/AP_Baro/AP_Baro_SPL06.cpp

```diff
@@ -103,9 +103,19 @@ bool AP_Baro_SPL06::_init()
     _dev->set_speed(AP_HAL::Device::SPEED_HIGH);
 
     uint8_t whoami;
-    if (!_dev->read_registers(SPL06_REG_CHIP_ID, &whoami, 1)  ||
-        whoami != SPL06_CHIP_ID) {
-        // not a SPL06
+
+    uint8_t retries=5;
```

Comment:
```suggestion
// Sometimes SPL06 has init problems, that's due to failure of reading using SPI for the first time. The SPL06 is a dual
// protocol sensor(I2C and SPI), sometimes it takes one SPI operation to convert it to SPI mode after it starts up.
```

(but probably line-wrapped differently)